src/Features/Core/Portable/IntroduceUsingStatement/AbstractIntroduceUsingStatementCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,285 @@
+                var parentStatements = GetStatements(declarationStatement.Parent);
+                var declarationStatementIndex = parentStatements.IndexOf(declarationStatement);
+
+                var newParent = WithStatements(
+                  declarationStatement.Parent,
+                  new SyntaxList<TStatementSyntax>(
+                      parentStatements.Take(declarationStatementIndex)
+                      .Concat(new[] { usingStatement })
+                      .Concat(parentStatements.Skip(declarationStatementIndex + 1 + statementsToSurround.Count))));
+
+                return document.WithSyntaxRoot(root.ReplaceNode(declarationStatement.Parent, newParent));
+            }
+            else
+            {
+                return document.WithSyntaxRoot(root.ReplaceNode(declarationStatement, usingStatement));
+            }
+        }
+
+        private SyntaxList<TStatementSyntax> GetStatementsToSurround(TLocalDeclarationSyntax declarationStatement, ILocalSymbol localVariable, SemanticModel semanticModel, ISyntaxFactsService syntaxFactsService, CancellationToken cancellationToken)
+        {
+            if (IsBlockLike(declarationStatement.Parent))
+            {
+                // Find the minimal number of statements to move into the using block
+                // in order to not break existing references to the local.
+                var lastUsageStatement = FindSiblingStatementContainingLastUsage(
+                    declarationStatement,
+                    localVariable,
+                    semanticModel,
+                    syntaxFactsService,
+                    cancellationToken);
+
+                if (lastUsageStatement != null)
+                {
+                    var parentStatements = GetStatements(declarationStatement.Parent);
+                    var declarationStatementIndex = parentStatements.IndexOf(declarationStatement);
+                    var lastUsageStatementIndex = parentStatements.IndexOf(lastUsageStatement, declarationStatementIndex + 1);
+
+                    return new SyntaxList<TStatementSyntax>(parentStatements
+                        .Take(lastUsageStatementIndex + 1)
+                        .Skip(declarationStatementIndex + 1));
+                }
+            }
+
+            return default;
+        }
+
+        private static (SyntaxTriviaList sameLine, SyntaxTriviaList endOfLine) SplitTrailingTrivia(SyntaxNode node, ISyntaxFactsService syntaxFactsService)
+        {
+            var trailingTrivia = node.GetTrailingTrivia();
+            var lastIndex = trailingTrivia.Count - 1;
+
+            return lastIndex != -1 && syntaxFactsService.IsEndOfLineTrivia(trailingTrivia[lastIndex])
+                ? (
+                    sameLine: trailingTrivia.RemoveAt(lastIndex),
+                    endOfLine: new SyntaxTriviaList(trailingTrivia[lastIndex]))
+                : (
+                    sameLine: trailingTrivia,
+                    endOfLine: SyntaxTriviaList.Empty);
```

Comment:
i particularly hate this formatting :)   it may be the `: (` frowny face.

Reply:
How would you format it?

Reply:
I'd be fine with:  

```c#
    ? (trailingTrivia.RemoveAt(lastIndex), new SyntaxTriviaList(trailingTrivia[lastIndex]))
    : (trailingTrivia, SyntaxTriviaList.Empty)
```